### DISCUSSION TITLE
Support replying to specific messages in a thread

### DISCUSSION THREAD

**Discussion by b-ggs** [Chatterino/chatterino2#4762] [Ideas]
### Checklist

- [X] I'm reporting a problem with Chatterino
- [X] I've verified that I'm running **the most** recent nightly build or stable release
- [X] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [X] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

On web chat, you can now reply to messages inside a thread, e.g.

```
user1: 1

┌ Replying to @user1: 1
user2: 2

┌ Replying to @user2: 2
user1: 3

┌ Replying to @user1: 3
user2: 4
```

However on Chatterino, right-clicking to reply to a message inside a thread sends a reply to the root of the thread, e.g.

```
user1: 1

┌ Replying to @user1: 1
user2: 2

┌ Replying to @user1: 1
user1: 3

┌ Replying to @user1: 1
user2: 4
```

I remember the former not being possible in web chat, so this seems to be a new feature.

### Screenshots

Below is a demonstration of replying to a message in a thread on web chat with no extensions vs Chatterino.

https://github.com/Chatterino/chatterino2/assets/6130147/39219929-4558-4002-93aa-8718b6a6c6bd

Can also confirm that this works with the FrankerFaceZ extension.

### OS and Chatterino Version

Chatterino 2.4.4 DEBUG (commit 5ee9697) built with Qt 5.15.10 Running on macOS 13.4, kernel: 22.5.0

**Comment by Felanbird** [comment]
This capability to implement this on our end is new-ish (a few weeks ago?) which is why we don't yet support it.

![image](https://github.com/Chatterino/chatterino2/assets/41973452/432c42ba-cf2c-4b6b-a1cb-1bed39fe75a3)
```
@first-msg=0;reply-thread-parent-msg-id=645c8c27-fbac-4341-a97e-19917d641e71;reply-parent-user-id=137690566;reply-parent-msg-id=79d4a26f-09c0-46f9-baad-deb8571c2a2a;tmi-sent-ts=1691736358777;turbo=0;user-id=22524109;display-name=Felanbird;subscriber=0;user-type=;badge-info=;client-nonce=2403985d3484bf917b051c28f3342efe;returning-chatter=0;color=#666666;reply-parent-msg-body=@Felanbird\s!!;reply-thread-parent-user-login=felanbird;room-id=11148817;mod=0;id=8a2d7bda-de02-4262-9831-78670f1bfd3f;reply-parent-user-login=magichackbot;badges=glhf-pledge/1;emotes=;reply-parent-display-name=MagicHackBot;flags= :felanbird!felanbird@felanbird.tmi.twitch.tv PRIVMSG #pajlada :@MagicHackBot buh
```

`reply-thread-parent-msg-id` `reply-parent-msg-id` both exist in these messages now, previously it was only `reply-parent-msg-id`

**Comment by hemirt** [comment]
still happens in the latest nightly 
Chatterino 2.4.5 (commit [783c7530](https://github.com/Chatterino/chatterino2/commit/783c7530)) built with Qt 5.12.12

**Reply by Felanbird** [reply]
that's probably because we didn't implement it yet

**Comment by b-ggs** [comment]
This has been implemented in #4923. Thanks all!
